D2L.Hypermedia.Siren/SirenEntity.cs

```diff
@@ -39,6 +43,17 @@ public class SirenEntity : ISirenEntity {
 			m_type = type;
 		}
 
+		public SirenEntity( SerializationInfo info, StreamingContext context ) :
+			this(
+				(string[])info.GetValue( "rel", typeof(string[]) ),
+				(string[])info.GetValue( "class", typeof(string[]) ),
+				JsonConvert.DeserializeObject( info.GetString( "properties" ), typeof(ExpandoObject) ),

```

Comment:
Does this still work for an anonymous object?

Reply:
it should for something like `properties:  new { foo = "bar"	},` (the test seems to serialize it back and forth)